### PR TITLE
[IDWA-OPS-58] Poetry test action for OCR

### PR DESCRIPTION
**Pull request by derekadombek**
# Pull Request

## Description
This will unit test as left and often as possible in CI in regards to work being done in OCR

## Related Issues
#58 

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [x] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [x] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [x] I have updated the documentation, if applicable.
- [x] I have added or updated test cases to cover my changes, if applicable.
- [x] I have minimized the number of reviewers to include only those essential for the review.
- [x] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: .github/workflows/ocr-tests.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: OCR Poetry Tests
+on:
+  pull_request:
+    types: [opened, reopened]
+    paths:
+      - .github/workflows/ocr-tests.yml
+      - OCR/**
```
Comment (derekadombek) [inline .github/workflows/ocr-tests.yml:7]: Do we want to make this scope even more narrow to just the `ocr` and `tests` directories within `OCR`?

Comment (zdeveloper) [reply .github/workflows/ocr-tests.yml:7]: no because we would still want to run the for dependency changes for example, pyproject.toml lives outside these two folders
my vote is the entire OCR directory

**Review by zdeveloper — APPROVED** [review]
LGTM! thank you! is there a way to test it before merging, I am cool with merging it and testing it afterwards, just curious

File: .github/workflows/ocr-tests.yml (hunk 1 above)
Comment (derekadombek) [reply .github/workflows/ocr-tests.yml:7]: For sure! I think thats the safe bet for now. Also, this is more in regards to when this CI will trigger. so for example, if you make a change in the `pyproject.toml` file and commit to your feature branch it will trigger this workflow.

**Comment by derekadombek** [comment]
> LGTM! thank you! is there a way to test it before merging, I am cool with merging it and testing it afterwards, just curious

You can! but you would need to pull the branch down locally and then add `branches: [idwa-OCR-58-poetry-test-action]` to `on: push` and then push the change to trigger the workflow but yeah it running in main and pushing a new branch would be easier. Also, there were some test run in the Actions tab.
